test/batch_notifier_test.py

```diff
@@ -0,0 +1,562 @@
+        bn.add_failure('Task(a=5)', 'Task', {'a': 5}, 'error')
+        bn.add_failure('Task(a=5)', 'Task', {'a': 5}, 'error')
+        bn.send_email()
+        self.check_email_send(
+            'Luigi: 3 failures in the last 60 minutes',
+            '- Task(a=5) (3 failures)'
+        )
+
+    def test_multiple_failures_of_multiple_jobs(self):
+        bn = BatchNotifier(batch_mode='all')
+        bn.add_failure('Task(a=5)', 'Task', {'a': 5}, 'error')
+        bn.add_failure('Task(a=6)', 'Task', {'a': 6}, 'error')
+        bn.add_failure('Task(a=6)', 'Task', {'a': 6}, 'error')
+        bn.send_email()
+        self.check_email_send(
+            'Luigi: 3 failures in the last 60 minutes',
+            '- Task(a=6) (2 failures)\n'
+            '- Task(a=5) (1 failure)'
+        )
+
+    def test_group_on_family(self):
+        bn = BatchNotifier(batch_mode='family')
+        bn.add_failure('Task(a=5)', 'Task', {'a': 5}, 'error')
+        bn.add_failure('Task(a=6)', 'Task', {'a': 6}, 'error')
+        bn.add_failure('Task(a=6)', 'Task', {'a': 6}, 'error')
+        bn.add_failure('OtherTask(a=6)', 'OtherTask', {'a': 6}, 'error')
+        bn.send_email()
+        self.check_email_send(
+            'Luigi: 4 failures in the last 60 minutes',
+            '- Task (3 failures)\n'
+            '- OtherTask (1 failure)'
+        )
+
+    def test_group_on_unbatched_params(self):
+        bn = BatchNotifier(batch_mode='unbatched_params')
+        bn.add_failure('Task(a=5, b=1)', 'Task', {'a': 5}, 'error')
+        bn.add_failure('Task(a=5, b=2)', 'Task', {'a': 5}, 'error')
+        bn.add_failure('Task(a=6, b=1)', 'Task', {'a': 6}, 'error')
+        bn.add_failure('Task(a=6, b=2)', 'Task', {'a': 6}, 'error')
+        bn.add_failure('Task(a=6, b=3)', 'Task', {'a': 6}, 'error')
+        bn.add_failure('Task(a=6, b=4)', 'Task', {'a': 6}, 'error')
+        bn.add_failure('OtherTask(a=5, b=1)', 'OtherTask', {'a': 5}, 'error')
+        bn.add_failure('OtherTask(a=6, b=1)', 'OtherTask', {'a': 6}, 'error')
+        bn.add_failure('OtherTask(a=6, b=2)', 'OtherTask', {'a': 6}, 'error')
+        bn.add_failure('OtherTask(a=6, b=3)', 'OtherTask', {'a': 6}, 'error')
+        bn.send_email()
+        self.check_email_send(
+            'Luigi: 10 failures in the last 60 minutes',
+            '- Task(a=6) (4 failures)\n'
+            '- OtherTask(a=6) (3 failures)\n'
+            '- Task(a=5) (2 failures)\n'
+            '- OtherTask(a=5) (1 failure)'
+        )
+
+    def test_include_one_expl(self):
+        bn = BatchNotifier(batch_mode='family', error_messages=1)
+        bn.add_failure('Task(a=1)', 'Task', {'a': 1}, 'error 1')
+        bn.add_failure('Task(a=2)', 'Task', {'a': 2}, 'error 2')
+        bn.add_failure('TaskB(a=1)', 'TaskB', {'a': 1}, 'error')
+
+        bn.send_email()
+        self.check_email_send(
+            'Luigi: 3 failures in the last 60 minutes',
+            '- Task (2 failures)\n'
+            '\n'
+            '      error 2\n'
```

Comment:
I updated the name of the unit test and added a short explanation to configuration.rst